catroid/src/main/java/org/catrobat/catroid/drone/jumpingsumo/JumpingSumoInitializer.java

```diff
@@ -0,0 +1,312 @@
+import com.parrot.arsdk.arcontroller.ARDeviceControllerStreamListener;
+import com.parrot.arsdk.arcontroller.ARFeatureCommon;
+import com.parrot.arsdk.arcontroller.ARFrame;
+import com.parrot.arsdk.ardiscovery.ARDISCOVERY_PRODUCT_ENUM;
+import com.parrot.arsdk.ardiscovery.ARDiscoveryDevice;
+import com.parrot.arsdk.ardiscovery.ARDiscoveryDeviceNetService;
+import com.parrot.arsdk.ardiscovery.ARDiscoveryDeviceService;
+import com.parrot.arsdk.ardiscovery.ARDiscoveryException;
+
+import org.catrobat.catroid.CatroidApplication;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.stage.PreStageActivity;
+import org.catrobat.catroid.stage.StageActivity;
+import org.catrobat.catroid.ui.dialogs.CustomAlertDialogBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.catrobat.catroid.CatroidApplication.getAppContext;
+
+public class JumpingSumoInitializer {
+
+	private static final List<ARDiscoveryDeviceService> DRONELIST = new ArrayList<>();
+	public JumpingSumoDiscoverer jsDiscoverer;
+
+	private ARDeviceController deviceController;
+	private static JumpingSumoInitializer instance;
+
+	private final Handler handler = new Handler(getAppContext().getMainLooper());
+
+	private static final String TAG = JumpingSumoInitializer.class.getSimpleName();
+
+	private PreStageActivity prestageStageActivity;
+	private StageActivity stageActivity = null;
+	private ARCONTROLLER_DEVICE_STATE_ENUM deviceState = ARCONTROLLER_DEVICE_STATE_ENUM.ARCONTROLLER_DEVICE_STATE_STOPPED;
+
+	private static final int JUMPING_SUMO_BATTERY_THRESHOLD = 3;
+	private static final int CONNECTION_TIME = 10000; //4000;
+	private static int jumpingSumoCount = 0;
+	private boolean messageShown = false;
+
+	public JumpingSumoInitializer() {
+	}
+
+	public static JumpingSumoInitializer getInstance() {
+		if (instance == null) {
+			instance = new JumpingSumoInitializer();
+		}
+		return instance;
+	}
+
+	public void setPreStageActivity(PreStageActivity prestageStageActivity) {
+		this.prestageStageActivity = prestageStageActivity;
+	}
+
+	public boolean disconnect() {
+		boolean success = false;
+		jsDiscoverer.removeListener(discovererListener);
+		if (deviceController != null) {
+			ARCONTROLLER_ERROR_ENUM error = deviceController.stop();
+			if (error == ARCONTROLLER_ERROR_ENUM.ARCONTROLLER_OK) {
+				success = true;
+				JumpingSumoDeviceController controller = JumpingSumoDeviceController.getInstance();
+				controller.setDeviceController(null);
+			}
+		}
+		return success;
+	}
+
+	public void initialise() {
+		jsDiscoverer = new JumpingSumoDiscoverer(getAppContext());
+		if (checkRequirements()) {
+			jsDiscoverer.setup();
+			jsDiscoverer.addListener(discovererListener);
+		}
+	}
+
+	public void checkJumpingSumoAvailability(PreStageActivity prestageStageActivityNow) {
+		setPreStageActivity(prestageStageActivityNow);
+		Log.d(TAG, "JumpSumo Count: " + jumpingSumoCount);
+
+		Handler handler = new Handler();
+		handler.postDelayed(new Runnable() {
```

Comment:
I tried what you wanted, but that is not possible. There is a Problem with connecting to the drone, if I change it.